lib/config-generator.js

```diff
@@ -0,0 +1,261 @@
+import { getShorthandName } from "./utils/naming.js";
+import * as log from "./utils/logging.js";
+
+// TODO: need to specify the package version - they may export flat configs in the future.
+const jsStyleGuides = [
+    { message: "Airbnb: https://github.com/airbnb/javascript", name: "airbnb", value: { packageName: "eslint-config-airbnb", type: "eslintrc" } },
+    { message: "Standard: https://github.com/standard/standard", name: "standard", value: { packageName: "eslint-config-standard", type: "eslintrc" } },
+    { message: "XO: https://github.com/xojs/eslint-config-xo", name: "xo", value: { packageName: "eslint-config-xo", type: "eslintrc" } }
+];
+const tsStyleGuides = [
+    { message: "Standard: https://github.com/standard/eslint-config-standard-with-typescript", name: "standard", value: { packageName: "eslint-config-standard-with-typescript", type: "eslintrc" } },
+    { message: "XO: https://github.com/xojs/eslint-config-xo-typescript", name: "xo", value: { packageName: "eslint-config-xo-typescript", type: "eslintrc" } }
+];
+
+/**
+ * Class representing a ConfigGenerator.
+ */
+export class ConfigGenerator {
+
+    /**
+     * Create a ConfigGenerator.
+     * @param {Object} options The options for the ConfigGenerator.
+     * @param {string} options.cwd The current working directory.
+     * @param {Object} options.answers The answers provided by the user.
+     * @returns {ConfigGenerator} The ConfigGenerator instance.
+     */
+    constructor(options = {}) {
+        this.cwd = options.cwd || process.cwd();
+        this.packageJsonPath = options.packageJsonPath || findPackageJson(this.cwd);
+        this.answers = options.answers || {};
+        this.result = {
+            devDependencies: ["eslint"],
+            configFilename: "eslint.config.js",
+            configContent: ""
+        };
+    }
+
+    /**
+     * Prompt the user for input.
+     * @returns {void}
+     */
+    async prompt() {
+        const questions = [
+            {
+                type: "select",
+                name: "purpose",
+                message: "How would you like to use ESLint?",
+                initial: 1,
+                choices: [
+                    { message: "To check syntax only", name: "syntax" },
+                    { message: "To check syntax and find problems", name: "problems" },
+                    { message: "To check syntax, find problems, and enforce code style", name: "style" }
+                ]
+            },
+            {
+                type: "select",
+                name: "moduleType",
+                message: "What type of modules does your project use?",
+                initial: 0,
+                choices: [
+                    { message: "JavaScript modules (import/export)", name: "esm" },
+                    { message: "CommonJS (require/exports)", name: "commonjs" },
+                    { message: "None of these", name: "script" }
+                ]
+            },
```

Comment:
`moduleType` is never used but it should be. If CommonJS is selected, it should set the `sourceType` of .js files accordingly, and it should add the `"commonjs"` globals in the `languageOptions`.

Reply:
addressed in https://github.com/eslint/create-config/pull/81/commits/bb8e042ed2f36f0a5637314f90411604e5551425.

it will set `languageOptions: {sourceType: "commonjs"}`, and commonjs globals will be auto-applied.

Reply:
The problem is that this will change the `sourceType` also for .cjs and .mjs files, making them both script or CommonJS depending on the choice. This seems incorrect since Node.js always treats those files as CommonJS and ESM respectively. It would be better to set the `sourceType` only for .js files. This was not a concern in eslintrc since it would only lint .js files by default.

Reply:
I see, will make a fix later!